octree/include/pcl/octree/octree_iterator.h

```diff
@@ -523,6 +523,89 @@ namespace pcl
         std::deque<IteratorState> FIFO_;
       };
 
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    /** \brief @b Octree iterator class
+     * \note Iterator  over all existing nodes at a given depth. It walks across an octree
+     *       in a breadth-first manner.
+     * \ingroup octree
+     * \author Fabien Rozar (fabien.rozar@gmail.com)
+     */
+    template<typename OctreeT>
+    class OctreeFixedDepthIterator : public OctreeBreadthFirstIterator<OctreeT>
+    {
+    public:
+
+      // public typedefs
+      using typename OctreeBreadthFirstIterator<OctreeT>::BranchNode;
+      using typename OctreeBreadthFirstIterator<OctreeT>::LeafNode;
+
+      /** \brief Empty constructor.
+        * \param[in] depth_arg Depth level during traversal
+        */
+      explicit
+      OctreeFixedDepthIterator (unsigned int depth_arg = 0);
+
+      /** \brief Constructor.
+      * \param[in] octree_arg Octree to be iterated. Initially the iterator is set to its root node.
+      * \param[in] depth_arg Depth level during traversal
+      */
+      explicit
+      OctreeFixedDepthIterator (OctreeT* octree_arg, unsigned int depth_arg = 0);
+
+      /** \brief Copy constructor.
+      * \param[in] octree_arg Octree to be iterated. Initially the iterator is set to its root node.
+      * \param[in] depth_arg Depth level during traversal
+      */
+      OctreeFixedDepthIterator (const OctreeFixedDepthIterator& iterator);
+
+      virtual
+      ~OctreeFixedDepthIterator ();
+
+      /** \brief Copy operator.
+      * \param[in] src the iterator to copy into this
+      */
+      inline OctreeFixedDepthIterator&
+      operator = (const OctreeFixedDepthIterator& src)
```

Comment:
I reintroduce the `operator =`, it make the `fixed_depth_` copy explicitly. I don't see an easy way to test it because the `operator==` (inherented for `OctreeIteratorBase`) never check the `fixed_depth_` value.

Should I add a specific `operator==` for the `OctreeFixedDepthIterator`? (In my opinion, this one from `OctreeIteratorBase` is enough, but in this case, I would like to see the explicit copy of the field  `fixed_depth_`.)